data-prepper-plugins/ruby-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/ruby/RubyProcessor.java

```diff
@@ -0,0 +1,201 @@
+            throw new RuntimeException(exception.toString());
+        }
+    }
+
+    private void assertThatInterfaceImplementedInRubyFile(InputStream fileStreamForTestingRegex)
+    throws IOException
+    {
+        String rubyCodeAsString = convertInputStreamToString(fileStreamForTestingRegex);
+
+        if (!stringContainsPattern(rubyCodeAsString, RUBY_METHOD_PROCESS_PATTERN)) {
+            throwBadInterfaceImplementationException(PROCESS_METHOD_NOT_FOUND_ERROR_MESSAGE);
+        }
+
+        this.fileCodeContainsInitCall = stringContainsPattern(rubyCodeAsString,
+                RUBY_METHOD_INIT_PATTERN);
+
+        if (!this.fileCodeContainsInitCall) {
+            String bareboneInitPattern = "def init";
+            boolean incorrectInitMethodSignature = stringContainsPattern(rubyCodeAsString, bareboneInitPattern);
+
+            if (incorrectInitMethodSignature) {
+                throwBadInterfaceImplementationException(BAD_INIT_METHOD_SIGNATURE_ERROR_MESSAGE);
+            }
+        }
+    }
+
+    private void throwBadInterfaceImplementationException(final String errorMessage) {
+        LOG.error(errorMessage);
+        throw new RuntimeException("Ruby Processor file bad implementation");
+    }
+
+    private boolean stringContainsPattern(String inString, String pattern) {
+        Pattern compiledPattern = Pattern.compile(pattern);
+        Matcher matcher = compiledPattern.matcher(inString);
+        return matcher.find();
+    }
+
+    private String convertInputStreamToString(InputStream inputStream)
+    throws IOException
+    {
+        return new String(inputStream.readAllBytes(), StandardCharsets.UTF_8);
+    }
+
+    private void runInitCodeFromFileAndDeclareTheProcessMethod() {
+        // Below will also pull in any dependencies in `require` statements.
+        container.runScriptlet(this.fileStream, "RUBY_FILE_FOR_LOGGING_PURPOSES");
+
+        if (this.fileCodeContainsInitCall) {
+            container.callMethod("", "init", config.getParams());
+        }
+    }
+
+    @Override
+    public Collection<Record<Event>> doExecute(final Collection<Record<Event>> records) {
+        final List<Event> events = records.stream().map(Record::getData).collect(Collectors.toList());
+        LOG.debug("Ruby Processor received events: [{}]", events);
+        injectAndProcessEvents(events);
+
+        return records;
+    }
+
+    private void injectAndProcessEvents(List<Event> events) {
+        container.put("events", events);
+        script = "events.each { |event| \n"
+                + this.codeToInject +
+                "}";
+
+        try {
+            container.runScriptlet(script);
+        } catch (Exception exception) {
+            LOG.error("Exception while processing Ruby code on Events: [{}]",events, exception);
+            if (!config.isIgnoreException()) {
+                throw new RuntimeException(format("Exception while processing Ruby code on Events: [%s]",events),
```

Comment:
Do we need to log all the Events? Would the Events be dropped if this exception is thrown? I think the preferred behavior would be to log the error and then return the Events so that they are sent through the pipeline like they were before they entered the ruby processor